Invalid memory address using AWS and IRSA

Hi folks!

I have an external-secret deployment version 0.3.6 running on an EKS cluster.

I have a `ClusterSecretStore` set to read from AWS Secrets Manager, configured as follows

```yaml
apiVersion: external-secrets.io/v1alpha1
kind: ClusterSecretStore
metadata:
  name: default
spec:
  provider:
    aws:
      auth:
        jwt:
          serviceAccountRef:
            name: external-secrets-aws-secrets-manager-store
      region: eu-west-1
      service: SecretsManager
---
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    eks.amazonaws.com/role-arn: <omitted role ARN>
  name: external-secrets-aws-secrets-manager-store
  namespace: external-secrets

```

When the operator pod starts, I am greeted with the following exception in the logs

```
I1007 10:32:47.965968       1 request.go:668] Waited for 1.003449755s due to client-side throttling, not priority and fairness, request: GET:https://172.20.0.1:443/apis/crd.k8s.amazonaws.com/v1alpha1?timeout=32s
{"level":"info","ts":1633602768.0200078,"logger":"controller-runtime.metrics","msg":"metrics server is starting to listen","addr":":8080"}
{"level":"info","ts":1633602768.020353,"logger":"setup","msg":"starting manager"}
I1007 10:32:48.020788       1 leaderelection.go:243] attempting to acquire leader lease external-secrets/external-secrets-controller...
{"level":"info","ts":1633602768.0208435,"logger":"controller-runtime.manager","msg":"starting metrics server","path":"/metrics"}
I1007 10:33:06.245208       1 leaderelection.go:253] successfully acquired lease external-secrets/external-secrets-controller
{"level":"info","ts":1633602786.2454455,"logger":"controller-runtime.manager.controller.externalsecret","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret","source":"kind source: /, Kind="}
{"level":"info","ts":1633602786.245523,"logger":"controller-runtime.manager.controller.externalsecret","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret","source":"kind source: /, Kind="}
{"level":"info","ts":1633602786.2455332,"logger":"controller-runtime.manager.controller.externalsecret","msg":"Starting Controller","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret"}
{"level":"info","ts":1633602786.2457767,"logger":"controller-runtime.manager.controller.secretstore","msg":"Starting EventSource","reconciler group":"external-secrets.io","reconciler kind":"SecretStore","source":"kind source: /, Kind="}
{"level":"info","ts":1633602786.2458222,"logger":"controller-runtime.manager.controller.secretstore","msg":"Starting Controller","reconciler group":"external-secrets.io","reconciler kind":"SecretStore"}
{"level":"info","ts":1633602786.3460972,"logger":"controller-runtime.manager.controller.secretstore","msg":"Starting workers","reconciler group":"external-secrets.io","reconciler kind":"SecretStore","worker count":1}
{"level":"info","ts":1633602786.3462548,"logger":"controller-runtime.manager.controller.externalsecret","msg":"Starting workers","reconciler group":"external-secrets.io","reconciler kind":"ExternalSecret","worker count":1}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x17b8ebf]

goroutine 300 [running]:
github.com/external-secrets/external-secrets/pkg/provider/aws/auth.sessionFromServiceAccount(0x2f09e90, 0xc000fad8f0, 0xc000feb6c0, 0x2f42578, 0xc000fd2f20, 0x2f25c48, 0xc0002ede50, 0xc0009eda8a, 0x6, 0x2bffa48, ...)
	/home/runner/work/external-secrets/external-secrets/pkg/provider/aws/auth/auth.go:163 +0x5ff
github.com/external-secrets/external-secrets/pkg/provider/aws/auth.New(0x2f09e90, 0xc000fad8f0, 0x2f42578, 0xc000fd2f20, 0x2f25c48, 0xc0002ede50, 0xc0009eda8a, 0x6, 0x2bffa50, 0x2bffa48, ...)
	/home/runner/work/external-secrets/external-secrets/pkg/provider/aws/auth/auth.go:75 +0x7ea
github.com/external-secrets/external-secrets/pkg/provider/aws.newClient(0x2f09e90, 0xc000fad8f0, 0x2f42578, 0xc000fd2f20, 0x2f25c48, 0xc0002ede50, 0xc0009eda8a, 0x6, 0x2bffa50, 0xc000fd2f20, ...)
	/home/runner/work/external-secrets/external-secrets/pkg/provider/aws/provider.go:50 +0xf2
github.com/external-secrets/external-secrets/pkg/provider/aws.(*Provider).NewClient(0x47226f0, 0x2f09e90, 0xc000fad8f0, 0x2f42578, 0xc000fd2f20, 0x2f25c48, 0xc0002ede50, 0xc0009eda8a, 0x6, 0x0, ...)
	/home/runner/work/external-secrets/external-secrets/pkg/provider/aws/provider.go:42 +0x99
github.com/external-secrets/external-secrets/pkg/controllers/externalsecret.(*Reconciler).Reconcile(0xc0008b4280, 0x2f09e90, 0xc000fad8f0, 0xc0009eda8a, 0x6, 0xc000e5f770, 0x21, 0xc000fad800, 0x0, 0x0, ...)
	/home/runner/work/external-secrets/external-secrets/pkg/controllers/externalsecret/externalsecret_controller.go:138 +0x6fb
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc000893ae0, 0x2f09de8, 0xc0008b4400, 0x26ccb60, 0xc000ee17a0)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:298 +0x30d
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc000893ae0, 0x2f09de8, 0xc0008b4400, 0x0)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:253 +0x205
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2(0xc000cc11d0, 0xc000893ae0, 0x2f09de8, 0xc0008b4400)
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:214 +0x6b
created by sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2
	/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.9.3/pkg/internal/controller/controller.go:210 +0x425
```

The cause seems to live somewhere here: https://github.com/external-secrets/external-secrets/blob/main/pkg/provider/aws/auth/auth.go#L163

Anyone got any pointer (pun not intended) on how to debug this issue?

Thanks!

--- a/pkg/provider/aws/auth/auth.go
+++ b/pkg/provider/aws/auth/auth.go
@@ -161,7 +161,7 @@ func sessionFromSecretRef(ctx context.Context, prov *esv1alpha1.AWSProvider, sto
 func sessionFromServiceAccount(ctx context.Context, prov *esv1alpha1.AWSProvider, store esv1alpha1.GenericStore, kube client.Client, namespace string, jwtProvider jwtProviderFactory) (*credentials.Credentials, error) {
 	if store.GetObjectKind().GroupVersionKind().Kind == esv1alpha1.ClusterSecretStoreKind {
 		if prov.Auth.JWTAuth.ServiceAccountRef.Namespace == nil {
-			return nil, fmt.Errorf("ServiceAccountRef has no Namespace field (mandatory for ClusterSecretStore specs)")
+			return nil, fmt.Errorf("serviceAccountRef has no Namespace field (mandatory for ClusterSecretStore specs)")
 		}
 		namespace = *prov.Auth.JWTAuth.ServiceAccountRef.Namespace
 	}

--- a/pkg/provider/aws/auth/auth_test.go
+++ b/pkg/provider/aws/auth/auth_test.go
@@ -434,7 +434,7 @@ func TestNewSession(t *testing.T) {
 					},
 				},
 			},
-			expectErr: "ServiceAccountRef has no Namespace field (mandatory for ClusterSecretStore specs)",
+			expectErr: "serviceAccountRef has no Namespace field (mandatory for ClusterSecretStore specs)",
 		},
 	}
 	for i := range rows {

